### PR TITLE
chore: optimize StatusMessageReply/StatusDateGroupLabel

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- optimizes StatusMessageReply and StatusDateGroupLabel 
- fixes a couple of issues with displayName being wrong in various places

Fixes: https://github.com/status-im/status-desktop/issues/9016

### Affected areas

StatusMessageReply, StatusDateGroupLabel

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Contact with a local nickname (Alexxx/Alex99), original msg and reply:
image

Correct (repeated) date group labels, same correct date formatting as in the msg tooltips:
image

### CODE REVIEW

File: src/app/modules/shared_models/message_item.nim [hunk 1]
```diff
@@ -146,7 +146,8 @@ proc initItem*(
     if result.quotedMessageAuthorAvatar == "":
       result.quotedMessageAuthorAvatar = quotedMessageDiscordMessage.author.avatarUrl
   else:
-    result.quotedMessageAuthorDisplayName = quotedMessageAuthorDetails.details.displayName
+    result.quotedMessageAuthorDisplayName = if quotedMessageAuthorDetails.details.localNickname != "": quotedMessageAuthorDetails.details.localNickname
```
[the remark this answers is not in the document]
Comment (caybro) [reply src/app/modules/shared_models/message_item.nim:149]: This is a regression that landed with the "quotedMessage*" changes

[the remark this answers is not in the document]
Comment (jrainville) [reply src/app/modules/shared_models/message_item.nim:149]: I thought `displayName` always chose the "right" one, ie the nickname if there is one. If I'm right, then the issue would be in the contacts service.

**Review by jrainville — COMMENTED** [review]
Looks good, but it would really simplify the frontend and others if `displayName` always was the **display** name. So the backend should give the right one using the logic you have of `nickname` first directly.

File: src/app/modules/shared_models/message_item.nim (hunk 1 above)
[the remark this answers is not in the document]
Comment (caybro) [reply src/app/modules/shared_models/message_item.nim:149]: It works correctly for the regular sender.displayName, no idea why it fails here; but it's really a regression, this used to work fine before

[the remark this answers is not in the document]
Comment (alexjba) [reply src/app/modules/shared_models/message_item.nim:149]: Looks like quotedMessageAuthorDetails.details.userDefaultDisplayName() should provide the right display name

File: ui/app/AppLayouts/Chat/panels/UserListPanel.qml [hunk 2]
```diff
@@ -96,7 +96,7 @@ Item {
                         messageContextMenu.isProfile = true
                         messageContextMenu.myPublicKey = userProfile.pubKey
                         messageContextMenu.selectedUserPublicKey = model.pubKey
-                        messageContextMenu.selectedUserDisplayName = model.displayName
+                        messageContextMenu.selectedUserDisplayName = nickName || userName
```
Comment (alexjba) [inline ui/app/AppLayouts/Chat/panels/UserListPanel.qml:99]: Might be safer to update in the model. Looks like there are more checks involved in choosing the right display name.

File: ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml [hunk 3]
```diff
@@ -78,55 +80,57 @@ Item {
...
+                            onClicked: root.replyProfileClicked(this, mouse)
                         }
                     }
                     StatusBaseText {
                         Layout.alignment: Qt.AlignVCenter
                         color: Theme.palette.baseColor1
                         font.pixelSize: Theme.secondaryTextFontSize
                         font.weight: Font.Medium
-                        text: replyDetails.amISender ? qsTr("You") : replyDetails.sender.displayName
+                        text: replyDetails.amISender ? qsTr("You") : replyDetails.sender.secondaryName || replyDetails.sender.displayName
```
Comment (alexjba) [inline ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml:91]: Shouldn't the display name be ok if we fix it in the model?

**Review by alexjba — COMMENTED** [review]
Looks good! WDYT of using only displayName in the UI and fix it in nim models? Looks like ContactsDto.userDefaultDisplayName() can provide what we need.

**Comment by caybro** [comment]
> Looks good! WDYT of using only displayName in the UI and fix it in nim models? Looks like ContactsDto.userDefaultDisplayName() can provide what we need.

I'll give it a shot on Monday 🙂

File: ui/StatusQ/src/StatusQ/Components/StatusDateGroupLabel.qml [hunk 4]
```diff
@@ -10,6 +9,10 @@ StatusBaseText {
     property double previousMessageTimestamp
     property double messageTimestamp
 
+    readonly property int msInADay: 86400000
+    readonly property int lastMessageInDays:  Math.floor(previousMessageTimestamp / msInADay)
```
Comment (micieslak) [inline ui/StatusQ/src/StatusQ/Components/StatusDateGroupLabel.qml:13]: this property is typed, so using `Math.floor` is not needed here at all

File: ui/StatusQ/src/StatusQ/Components/StatusDateGroupLabel.qml [hunk 5]
```diff
@@ -10,6 +9,10 @@ StatusBaseText {
...
+    readonly property int currentMessageInDays: Math.floor(messageTimestamp / msInADay)
```
Comment (micieslak) [inline ui/StatusQ/src/StatusQ/Components/StatusDateGroupLabel.qml:14]: same here

File: ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml [hunk 6]
```diff
@@ -78,55 +80,57 @@ Item {
...
-                    textField.font.pixelSize: Theme.secondaryTextFontSize
-                    textField.color: Theme.palette.baseColor1
-                    visible: !!replyDetails.messageText && replyDetails.contentType !== StatusMessage.ContentType.Sticker
-                    allowShowMore: false
-                    stripHtmlTags: true
-                    convertToSingleLine: true
-                    messageDetails: root.replyDetails
-                }
-                StatusImageMessage {
+                Loader {
```
Comment (micieslak) [inline ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml:94]: could it be set asynchronous?

**Review by micieslak — COMMENTED** [review]
Looks ok, some minor comments.

I noticed that `StatusMessageReply` uses `Shape` component. From performance pov it may be a problem, we probably should take a closer look at that. It should be quite easy to replace it with static image + some rectangles.

File: src/app/modules/shared_models/message_item.nim (hunk 1 above)
[the remark this answers is not in the document]
Comment (caybro) [reply src/app/modules/shared_models/message_item.nim]: Works like a charm, thanks :)

File: ui/app/AppLayouts/Chat/panels/UserListPanel.qml (hunk 2 above)
Comment (caybro) [reply ui/app/AppLayouts/Chat/panels/UserListPanel.qml]: Well this is a different model and it has to display both (nickname and displayName); corrected to use the list item's `title` as that contains the required logic already

File: ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml (hunk 3 above)
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml]: Yup, correct; removed

File: ui/StatusQ/src/StatusQ/Components/StatusDateGroupLabel.qml (hunk 4 above)
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Components/StatusDateGroupLabel.qml]: Done

File: ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml (hunk 6 above)
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Components/private/statusMessage/StatusMessageReply.qml:94]: Yup, done

**Review by alexjba — APPROVED** [review]
Nice! 👍

**Review by micieslak — APPROVED** [review]
:+1:

**Review by noeliaSD — APPROVED** [review]
LGTM!

**Pull request merged**